cmd/juju/model/configcommand.go

```diff
@@ -301,43 +301,63 @@ func (c *configCommand) setConfig(client configCommandAPI, ctx *cmd.Context) err
 // get writes the value of a single key or the full output for the model to the cmd.Context.
 func (c *configCommand) getConfig(client configCommandAPI, ctx *cmd.Context) error {
 	if len(c.keys) == 1 && certBytes != nil {
-		ctx.Stdout.Write(certBytes)
+		_, _ = ctx.Stdout.Write(certBytes)
 		return nil
 	}
-	attrs, err := client.ModelGetWithMetadata()
+	attrs, err := c.getSkimmedModel(client)
 	if err != nil {
 		return err
 	}
+	attrs, err, finished := c.handleIsKeyOfModel(attrs, ctx)
+	if err != nil {
+		return err
+	} else if attrs != nil && finished {
+		return c.out.Write(ctx, attrs)
+	} else if len(c.keys) > 0 && !finished {
+		if isFileLike(c.keys[0]) {
```

Comment:
@mitechie so I was not 100% sure if there can be cases where the "keyname" could actually be looking like a yml file. That's why I did the check here and not above in https://github.com/juju/juju/pull/10665/files#diff-fc38c887834ec43ba3bdea51be58fd06L157-R160

I do think that, if the keyname could not be fileLike, I would move it from here to above. So ordering makes a little more sense in this flow.